Fill.py

```diff
@@ -15,8 +17,71 @@ class FillError(RuntimeError):
     pass
 
 
-def _log_fill_progress(name: str, placed: int, total_items: int) -> None:
-    logging.info(f"Current fill step ({name}) at {placed}/{total_items} items placed.")
+class FillLogger():
+    min_size: int = 1000   # minimum number of items we care about reporting (non-tty)
```

Comment:
probably a good idea to put these comments in docstring instead. it's not something we do strongly, but something i've been trying to do more at least. there might be reason to reuse this class later.